### DISCUSSION TITLE
somethings changed after upgrading megatinycore

### DISCUSSION THREAD

**Discussion by br74649** [SpenceKonde/megaTinyCore#646] [Q&A]
Hi, 

I have a library for reading rc pwm https://github.com/simondlevy/RXInterrupt

yes it uses attachinterrupt but i just drop it in and it used to work great for my hacking projects on the attiny1614

today i upgraded from 2.4.2 to the latest megatinycore 2.5.10 (deleted the folders as per the readme) and it now just returns "24" as the pwm signal on PIN_PA5, and some very strange numbers on PIN_PB1

Do i have to bite the bullet and use direct port stuff or is there a simpler (for me) resolution to this

TIA 

Tony

my code with the above RXinterrupt.cpp and RXinterrupt.h as an example is

#include "RXinterrupt.h"

void setup() {
 Serial.begin(9600);
 Serial.println("one");
 int pins[1] = {PIN_PA5};
 initChannels(pins, 1);
}

void loop() {
  short values[1];
  updateChannels(values, 1);
  Serial.println(values[0]);
  delay(20);
};

**Comment by SpenceKonde** [comment]
Choose attachInterrupt mode -> old version, there seem to still be sme issues, and right now I need to get something out to the poor ATTinyCore users who haven;t had a release in over a year.

**Reply by br74649** [reply]
Spot on, your know your core :) 

thats awesome...  made it work on the 1624 chips too

in platformio (which i know you dont use) i just had to add 

build_flags = 
  -DCORE_ATTACH_OLD

and all my stuff is working again

there has been activity recently on updating platformio to support the avr-2 chips which is why im down this rabbit hole in the first place

Thanks for your assistance 

Tony

**Comment by SpenceKonde** [comment]
If you could replace WInterrupt.c with the version from github, and let me know if it fixes your problem that would be much appreciated. I found a place where I had reversed two operands to a store instruction while preparing a pointer, and the result would explain why it was breaking so badly: 2 clocks later, it took a pointer which now pointed to whatever the user function happened to leave in r26 (I zero'ed r27). This could include any VPORT register, the unprotected SLPCTRL, RSTCTRL and CLKCTRL registers, the writable portions of the BOD registers, the VREF configuration registers, as well as the stack pointer itself, which would result in the restoration of the old register values restoringthe wrong values, The SREG could also be clobbered, but as there were no more tests before that was restored to it's value before the ISR fired, that was no worse than if it was hitting read only or non-writable fields, though I would expect  execution to slow to a crawl until the intflags got hit by sheer dumb luck.
